src/Help/TicketSystem/TicketRepository.php

```diff
@@ -0,0 +1,342 @@
+    private TicketMessageRepository $messageRepo;
+    private MessageRepository $userMessageRepo;
+
+    private int $inactiveTime = 72 * 3600; // 72 hours
+
+    public function __construct(
+        Connection $connection,
+        AdminUserRepository $adminUserRepo,
+        TicketMessageRepository $messageRepo,
+        MessageRepository $userMessageRepo
+    ) {
+        parent::__construct($connection);
+        $this->adminUserRepo = $adminUserRepo;
+        $this->messageRepo = $messageRepo;
+        $this->userMessageRepo = $userMessageRepo;
+    }
+
+    public function countNew(): int
+    {
+        return (int) $this->createQueryBuilder()
+            ->select("COUNT(*)")
+            ->from('tickets')
+            ->where("status = 'new'")
+            ->execute()
+            ->fetchOne();
+    }
+
+    public function countAssigned(int $adminId): int
+    {
+        if ($adminId == 0) {
+            return 0;
+        }
+        return (int) $this->createQueryBuilder()
+            ->select("COUNT(*)")
+            ->from('tickets')
+            ->where("status = 'assigned'")
+            ->andWhere('admin_id = :admin_id')
+            ->setParameter('admin_id', $adminId)
+            ->execute()
+            ->fetchOne();
+    }
+
+    public function find(int $id): ?Ticket
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('tickets')
+            ->where('id = :id')
+            ->setParameter('id', $id)
+            ->execute()
+            ->fetchAssociative();
+        return $data ? $this->createObject($data) : null;
+    }
+
+    public function findBy(array $args = [])
+    {
+        $qry = $this->createQueryBuilder()
+            ->select("*")
+            ->from('tickets')
+            ->orderBy('status')
+            ->addOrderBy('timestamp', 'DESC');
+        foreach ($args as $key => $val) {
+            $qry->andWhere($key . ' = :' . $key)
+                ->setParameter($key, $val);
+        }
+        $data = $qry->execute()
+            ->fetchAllAssociative();
+
+        return collect($data)
+            ->map(fn ($arr) => $this->createObject($arr))
+            ->toArray();
+    }
+
+    private function createObject(array $data): Ticket
+    {
+        $ticket = new Ticket();
+
+        $ticket->id = (int) $data['id'];
+        $ticket->solution = $data['solution'];
+        $ticket->status = $data['status'];
+        $ticket->timestamp = (int) $data['timestamp'];
+        $ticket->adminComment = $data['admin_comment'];
+        $ticket->catId = (int) $data['cat_id'];
+        $ticket->userId = (int) $data['user_id'];
+        $ticket->adminId = (int) $data['admin_id'];
+
+        return $ticket;
+    }
+
+    public function persist(Ticket &$ticket): bool
+    {
+        assert(isset(Ticket::STATUS_ITEMS[$ticket->status]), 'Invalid ticket status');
+        assert(isset(Ticket::SOLUTION_ITEMS[$ticket->solution]), 'Invalid ticket solution');

```

Comment:
I don't think assert does anything unless we adjust the php.ini

Validation should ideally happen outside the repository

Reply:
Solved as part of 331290b